app/views/layouts/application.html.erb

```diff
@@ -2,57 +2,79 @@
 <html>
   <head>
     <title>NameofApp</title>
-      
-    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">  
+
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
 
     <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
-      
-    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/js/bootstrap.min.js"></script>  
-      
+
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/js/bootstrap.min.js"></script>
+
     <meta name="viewport" content="width=device-width, initial-scale=1">
-     <%= csrf_meta_tags %>
+    <%= csrf_meta_tags %>
   </head>
 
   <body>
-    <nav class="navbar navbar-inverse navbar-fixed-top navbar-right" role="navigation">
-      <ul>
-         <li><%= link_to "Contact",  simple_pages_contact_path %></li>
-         <li><%= link_to "About", simple_pages_about_path %></li>
-         <li><%= link_to "Home", root_path %></li>
-     </ul>
-     <ul class="nav pull-right">
-       <li>
-       <% if user_signed_in? %>
-       <%= link_to('Logout', destroy_user_session_path, method: :delete) %>
-       <% else %>
-       <%= link_to('Login', new_user_session_path) %>
-       <% end %>
-       </li>
-      </ul>
-        
-        <%= form_with(url: products_path, method: 'get', local: true) do |form| %>
-         <%= form.label :q, 'Search for:' %>
-         <%= form.text_field :q, id: 'q', value: params[:q] %>
-         <%= form.submit 'Search' %>
-        <% end %>
-        
```

Comment:
Your custom navbar had some issues in different viewports, I replaced it with the bootstrap navbar which is mobile friendly.